### PR TITLE
Don't run static bashrc only for pure nix-shell

### PR DESCRIPTION
**Pull request by kubek2k**

### CODE REVIEW

**Comment by LnL7** [comment]
This was added as a workaround in the first place, do you have any idea why this doesn't seem necessary on linux?

**Comment by kubek2k** [comment]
> This was added as a workaround in the first place, do you have any idea why this doesn't seem necessary on linux?

I have no idea - since I don't use NixOS (or nix on any other linux). Maybe it could be a versioning issue (bash v2.7 on mac as default, as oppposed to v4 everywhere else?)

**Comment by kubek2k** [comment]
@LnL7 the nix-shell man states that the `IN_NIX_SHELL` started to be set since nix 2.0. Maybe that was the cause?

**Comment by hagl** [comment]
Is there a chance to get this PR merged? 
I use `programs.bash.interactiveShellInit` in my `darwin-configuration.nix` to setup a shell prompt containing the git branch, but that doesn't work in `nix-shell`s because of this early return.

File: modules/programs/bash/default.nix [hunk 1]
```diff
@@ -59,7 +59,7 @@ in
       __ETC_BASHRC_SOURCED=1
 
       # Don't execute this file when running in a pure nix-shell.
-      if test -n "$IN_NIX_SHELL"; then return; fi
+      if [ "$IN_NIX_SHELL" == "pure" ]; then return; fi
```
Comment (Enzime) [inline modules/programs/bash/default.nix:62]:
```suggestion
      if [ "$IN_NIX_SHELL" = "pure" ]; then return; fi
```

**Review by Enzime — CHANGES_REQUESTED** [review]
Thanks for the contribution, can you rebase your PR?

**Comment by Enzime** [comment]
@kubek2k can you rebase the PR? I can't merge until all the tests have succeeded

**Review by Enzime — APPROVED** [review]
LGTM

**Pull request merged**